src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    stateCopy = { ...stateCopy };
```

Comment:
you don't need to copy the stateCopy

Reply:
Without it, it doesn't pass my tests, this is so that it assigns each time to a new object and not to the reference of the same one. I'm not sure if I understand it correctly, but I was looking for a solution and I found this one.

Reply:
how about doing the only copy in the `for` instead of above it.

Reply:
First, I create the first copy so as not to modify the input array, then I make a copy in the loop each time so that modifications in one loop do not affect the result added to result through references, without it it fails 2 tests.

  × Should handle multiple types (5ms)                                                                                                                                                                        
  × Should handle a long list of types (1ms)  
  
  If you add "let stateCopy = { ...state };" to the loop, each object in the result array would be identical. If I add this code to the loop, it also fails the tests.